src/components/childComponents/CreateAndUpload.js

```diff
@@ -0,0 +1,150 @@
+	let response = await fetch(upload_to_dataset_url, {
+		method: "POST",
+		mode: "cors",
+		headers: authHeader,
+		body: body,
+	});
+
+	if (response.status === 200) {
+		// return file ID
+		// {id:xxx} OR {ids:[{id:xxx}, {id:xxx}]}
+		return response.json();
+	} else if (response.status === 401) {
+		// TODO handle error
+		return {};
+	} else {
+		// TODO handle error
+		return {};
+	}
+}
+
+async function extractionsRequest(file,body_data) {
+	const file_id = file["id"];
+	const extractions_url = `${config.hostname}/clowder/api/files/${file_id}/extractions`;
+	const body = JSON.stringify(body_data);
+	let authHeader = getHeader();
+	authHeader.append('Accept', 'application/json');
+	authHeader.append('Content-Type', 'application/json');
+	let response = await fetch(extractions_url, {
+		method: "POST",
+		mode: "cors",
+		headers: authHeader,
+		body: body,
+	});
+
+	if (response.status === 200) {
+		// return {"status":"OK","job_id":"string"}
+		//return response.json();
+		console.log(response);
+	} else if (response.status === 401) {
+		// TODO handle error
+		console.log("error");
+		//return {};
+	} else {
+		// TODO handle error
+		console.log("error");
+		//return {};
+	}
+}
+
+export default function CreateAndUpload() {
+	const [dropFile, setDropFile] = useState([]); // state for dropped file
+	const [clowderFile, setClowderFile] = useState(null);  // state for uploaded file in Clowder
+
+	// if dropFile state has changed, create and upload to dataset
+	useEffect(async () => {
+		const name = dropFile.name;
+		const description = dropFile.type;
+		if (name !== undefined) {
+			const body = {"name": name, "description": description};
+			const dataset = await createDatasetRequest(body);
+			if (dataset["id"] !== undefined) {
+				await uploadToDatasetRequest(dataset["id"], dropFile).then((response) => {setClowderFile(response)} );
+			}
+		}
+		else {
+			console.log("error in dropped file");
+		}
+	}, [dropFile]);
+
+	// if clowderFile state has changed, submit file for extraction. if disabled by default.
+	useEffect(async () => {
+		if (clowderFile !== null) {
+			const body = {"extractor": "ncsa.rctTransparencyExtractor"};
+			//const extractor_name = "ncsa.wordcount";
+			await extractionsRequest(clowderFile, body);
+
+		}
+	}, [clowderFile]);
+
+	// onDrop function
+	const onDrop = useCallback(acceptedFiles => {
+		// this callback will be called after files get dropped, we will get the acceptedFiles. If you want, you can even access the rejected files too
+		acceptedFiles.map(file => setDropFile(file));
+	}, []);
+	// TODO have a dependancy here - mouse hover or dropped file action
+
+	// We pass onDrop function and accept prop to the component. It will be used as initial params for useDropzone hook
+	return (
+		<Box className="createupload">
+			<Dropfile onDrop={onDrop}
+					  accept={ {'image/png': ['.png'], 'image/jpeg': ['.jpg', '.jpeg'], 'text/html': ['.html', '.htm'], 'text/plain':['.txt']} }
```

Comment:
Should we change this to accept just text/plain for now?

Reply:
updated to accepting just plain text files